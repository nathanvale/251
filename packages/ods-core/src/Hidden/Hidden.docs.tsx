/* eslint-disable react/display-name */
import React from "react";
import { Placeholder, Stack } from "@origin-digital/ods-core";
import { ComponentDocs } from "@origin-digital/ods-types";
import { Hidden, HiddenProps } from "../Hidden/Hidden";

export const docs: ComponentDocs<HiddenProps> = {
  category: "Layout",
  componentName: "Hidden",
  description:
    "Responsively hides children based on the selected implementation.",
  propDescriptions: {
    below: "Hides content below the selected breakpoint",
    above: "Hides content above the selected breakpoint",
    inline:
      "If true the <Hidden/> container is set to a span instead of a div. This is useful for hiding text in text nodes.",
  },
  migrationGuide: false,
  examples: {
    default: {
      Code: () => (
        <Hidden>
          <Placeholder label="Not hidden" />
        </Hidden>
      ),
    },
    additional: [
      {
        label: "Hiding components below the md breakpoint",
        Code: () => (
          <Stack space="small">
            <Placeholder label="The following line is hidden below md:" />
            <Hidden below="md">
              <Placeholder label="Hidden below md" />
            </Hidden>
          </Stack>
        ),
      },
      {
        label: "Hiding components above the sm breakpoint",
        Code: () => (
          <Stack space="small">
            <Placeholder label="The following line is hidden above sm:" />
            <Hidden above="sm">
              <Placeholder label="Hidden above sm" />
            </Hidden>
          </Stack>
        ),
      },
      {
        label:
          "Hiding components above the sm breakpoint and below to lg breakpoint",
        Code: () => (
          <Stack space="small">
            <Placeholder label="The following line is hidden above sm and below lg:" />
            <Hidden above="sm" below="lg">
              <Placeholder label="Hidden above sm and below lg" />
            </Hidden>
          </Stack>
        ),
      },
      {
        label: "Hiding text nodes below md",
        Code: () => (
          <span>
            The following text node is hidden below md:
            <Hidden below="md" inline={true}>
              Hidden below md.
            </Hidden>
          </span>
        ),
      },
      {
        label: "Hiding text nodes above sm",
        Code: () => (
          <span>
            The following text node is hidden above sm:
            <Hidden above="sm" inline={true}>
              Hidden above sm.
            </Hidden>
          </span>
        ),
      },
    ],
  },
  snippets: [],
};
