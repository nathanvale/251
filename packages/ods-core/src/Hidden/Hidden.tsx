import React, { ReactNode } from "react";
import { Box } from "@origin-digital/ods-core";
import { resolveResponsiveRangeProps } from "@origin-digital/ods-helpers";

import { ResponsiveRangeProps } from "@origin-digital/ods-types";

export interface HiddenProps extends ResponsiveRangeProps {
  children: ReactNode;
  inline?: boolean;
}

export const Hidden = ({ children, above, below, inline }: HiddenProps) => {
  const [
    hiddenOnXs,
    hiddenOnSm,
    hiddenOnMd,
    hiddenOnLg,
    hiddenOnXl,
  ] = resolveResponsiveRangeProps({
    above,
    below,
  });

  const display = inline ? "inline" : "block";

  return (
    <Box
      display={{
        xs: hiddenOnXs ? "none" : display,
        sm: hiddenOnSm ? "none" : display,
        md: hiddenOnMd ? "none" : display,
        lg: hiddenOnLg ? "none" : display,
        xl: hiddenOnXl ? "none" : display,
      }}
      component={inline ? "span" : "div"}
    >
      {children}
    </Box>
  );
};

Hidden.displayName = "Hidden";

Hidden.defaultProps = {
  inline: false,
};
