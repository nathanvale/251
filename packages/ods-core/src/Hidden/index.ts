export { Hidden } from "./Hidden";
export type HiddenProps = import("./Hidden").HiddenProps;
