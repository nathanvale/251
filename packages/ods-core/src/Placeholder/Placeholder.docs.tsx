import React from "react";
import { ComponentDocs } from "@origin-digital/ods-types";
import { Placeholder, PlaceholderProps } from "./Placeholder";

export const docs: ComponentDocs<PlaceholderProps> = {
  category: "Layout",
  componentName: "Placeholder",
  description:
    "This component is helpful for laying out placeholders for future content.",
  propDescriptions: {},
  migrationGuide: false,
  examples: {
    default: {
      Code: () => <Placeholder />,
    },
    additional: [
      {
        label: "Setting the Label",
        Code: () => <Placeholder label="Placeholder for content" />,
      },
      {
        label: "Setting the Shape",
        Code: () => <Placeholder shape="round" width={120} height={120} />,
      },
      {
        label: "Setting the Width",
        Code: () => <Placeholder width={120} />,
      },
      {
        label: "Setting the Height",
        Code: () => <Placeholder height={400} />,
      },
      {
        label: "Responsive Height Across Mobile and Desktop",
        description:
          "Width can be controlled responsively by passing in an array of mobile and desktop pixel sizes as values. Open in Playroom to see the responsive behaviour of this example.",
        Code: () => <Placeholder height={[100, 500]} />,
      },
      {
        label: "Responsive Height Across All Five Breakpoints",
        description:
          "For more granular control across all 5 breakpoints, ‘xs’, ‘sm’,  ‘md’, ‘lg’ and ‘xl’, width can be controlled responsively by passing in an object with breakpoints as keys and pixel sizes as values. Open in Playroom to see the responsive behaviour of this example.",
        Code: () => (
          <Placeholder
            height={{
              xs: 100,
              sm: 200,
              md: 300,
              lg: 400,
              xl: 500,
            }}
          />
        ),
      },
    ],
  },
  snippets: [],
};
