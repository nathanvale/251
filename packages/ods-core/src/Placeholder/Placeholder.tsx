import React from "react";
import styled, { css } from "styled-components";
import { width, height, WidthProps, HeightProps } from "styled-system";
import { ResponsiveProp } from "@origin-digital/ods-types";
import { normaliseResponsiveProp } from "@origin-digital/ods-helpers";
import { Box, BoxProps } from "../Box/Box";

type PlaceholderShapeVariant = "rectangle" | "round";
export interface PlaceholderProps {
  "data-id"?: string;
  height?: ResponsiveProp<string | number>;
  label?: string;
  shape?: PlaceholderShapeVariant;
  width?: ResponsiveProp<string | number>;
}

interface ContainerProps extends WidthProps, HeightProps {
  shape: PlaceholderProps["shape"];
}

const Container = styled<Omit<BoxProps, "width" | "height">>(Box)<
  ContainerProps
>`
  ${width}
  ${height}
  border: 2px solid rgba(51, 51, 51, 0.3);
  ${p =>
    p.shape === "round"
      ? css`
          border-radius: 50%;
        `
      : undefined}}
`;

const defaultHeight = 120;
const defaultWidth = "auto";
const defaultShape = "rectangle";

export const Placeholder = ({
  "data-id": dataId,
  label,
  width = defaultWidth,
  height = defaultHeight,
  shape = defaultShape,
}: PlaceholderProps) => {
  return (
    <Container
      data-id={dataId}
      position="relative"
      overflow="hidden"
      display="flex"
      alignItems="center"
      justifyContent="center"
      width={normaliseResponsiveProp<string | number>(width)}
      height={normaliseResponsiveProp<string | number>(height)}
      shape={shape}
      backgroundColor="grey16"
    >
      {label ? (
        <Box
          paddingX="xsmall"
          paddingY="xxsmall"
          textAlign="center"
          style={{ marginTop: "2px" }}
        >
          {label}
        </Box>
      ) : (
        <svg
          xmlns="http://www.w3.org/2000/svg"
          style={{ width: "100%", height: "100%", position: "absolute" }}
        >
          <line
            style={{ strokeWidth: "2px", stroke: "rgba(51,51,51,.1)" }}
            x1={0}
            y1={0}
            x2="100%"
            y2="100%"
          />
          <line
            style={{ strokeWidth: "2px", stroke: "rgba(51,51,51,.1)" }}
            x1="100%"
            y1={0}
            x2={0}
            y2="100%"
          />
        </svg>
      )}
    </Container>
  );
};

Placeholder.defaultProps = {
  "data-id": "placeholder",
  height: defaultHeight,
  inline: false,
  shape: defaultShape,
  width: defaultWidth,
};

Placeholder.displayName = "Placeholder";
