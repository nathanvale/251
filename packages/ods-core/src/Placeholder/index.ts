export { Placeholder } from "./Placeholder";
export type PlaceholderProps = import("./Placeholder").PlaceholderProps;
