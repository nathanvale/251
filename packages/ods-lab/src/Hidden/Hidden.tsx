import React, {ReactNode} from "react";
import {Box} from "@origin-digital/ods-core";
import {
  resolveResponsiveRangeProps,
  ResponsiveRangeProps,
} from "@origin-digital/ods-helpers";
import styled, {css} from "styled-components";

export interface HiddenProps extends ResponsiveRangeProps {
  children: ReactNode;
  screen?: boolean;
  print?: boolean;
  inline?: boolean;
}

const StyledBox = styled(Box)<{hiddenOnPrint: boolean}>`
  ${p =>
    p.hiddenOnPrint
      ? css`
          @media print {
            display: none !important;
          }
        `
      : ``}
`;

export const Hidden = ({
  children,
  above,
  below,
  screen,
  print,
  inline,
}: HiddenProps) => {
  const hiddenOnScreen = Boolean(screen);
  const hiddenOnPrint = Boolean(print);

  const [
    hiddenOnMobile,
    hiddenOnTablet,
    hiddenOnDesktop,
  ] = resolveResponsiveRangeProps({
    above,
    below,
  });

  const display = inline ? "inline" : "block";

  return (
    <StyledBox
      hiddenOnPrint={hiddenOnPrint}
      display={
        hiddenOnScreen
          ? "none"
          : {
              xs: hiddenOnMobile ? "none" : display,
              sm: hiddenOnMobile ? "none" : display,
              md: hiddenOnTablet ? "none" : display,
              lg: hiddenOnDesktop ? "none" : display,
              xl: hiddenOnDesktop ? "none" : display,
            }
      }
      component={inline ? "span" : "div"}
    >
      {children}
    </StyledBox>
  );
};

Hidden.displayName = "Hidden";

Hidden.defaultProps = {
  screen: false,
  print: false,
  inline: false,
};
