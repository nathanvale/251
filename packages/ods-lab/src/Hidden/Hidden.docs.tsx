/* eslint-disable react/display-name */
import React from "react";
import {Placeholder, Stack} from "@origin-digital/ods-core";
import {ComponentDocs} from "../../../../docs/src/types";
import {Hidden, HiddenProps} from "..";

export const docs: ComponentDocs<HiddenProps> = {
  category: "Experimental",
  componentName: "Hidden",
  description:
    "Responsively hides children based on the selected implementation.",
  propDescriptions: {
    screen: "If true, hides content on screen readers",
    print: "If true, hides content when printing",
    below: "Hides content below the selected breakpoint",
    above: "Hides content above the selected breakpoint",
    inline: "If true, hides inline text nodes",
  },
  migrationGuide: false,
  examples: [
    {
      label: "Hiding components below the tablet breakpoint",
      Code: () => (
        <Stack space="small">
          <Placeholder>The following line is hidden below tablet:</Placeholder>
          <Hidden below="tablet">
            <Placeholder>Hidden below tablet.</Placeholder>
          </Hidden>
        </Stack>
      ),
    },
    {
      label: "Hiding components below the desktop breakpoint",
      Code: () => (
        <Stack space="small">
          <Placeholder>The following line is hidden below desktop:</Placeholder>
          <Hidden below="desktop">
            <Placeholder>Hidden below desktop.</Placeholder>
          </Hidden>
        </Stack>
      ),
    },
    {
      label: "Hiding components above the mobile breakpoint",
      Code: () => (
        <Stack space="small">
          <Placeholder>The following line is hidden above mobile:</Placeholder>
          <Hidden above="mobile">
            <Placeholder>Hidden above mobile.</Placeholder>
          </Hidden>
        </Stack>
      ),
    },
    {
      label: "Hiding components above the tablet breakpoint",
      Code: () => (
        <Stack space="small">
          <Placeholder>The following line is hidden above tablet:</Placeholder>
          <Hidden above="tablet">
            <Placeholder>Hidden above tablet.</Placeholder>
          </Hidden>
        </Stack>
      ),
    },
    {
      label: "Hiding components when on print",
      Code: () => (
        <Stack space="small">
          <Placeholder>The following line is hidden on print:</Placeholder>
          <Hidden print={true}>
            <Placeholder>Hidden on print.</Placeholder>
          </Hidden>
        </Stack>
      ),
    },
    {
      label: "Hiding components when on screen readers",
      Code: () => (
        <Stack space="small">
          <Placeholder>The following line is hidden on screen:</Placeholder>
          <Hidden screen={true}>
            <Placeholder>Hidden on screen.</Placeholder>
          </Hidden>
        </Stack>
      ),
    },
    {
      label: "Hiding text nodes below tablet",
      Code: () => (
        <Placeholder>
          The following text node is hidden below tablet:
          <Hidden below="tablet" inline={true}>
            Hidden below tablet.
          </Hidden>
        </Placeholder>
      ),
    },
    {
      label: "Hiding text nodes below desktop",
      Code: () => (
        <Placeholder>
          The following text node is hidden below desktop:
          <Hidden below="desktop" inline={true}>
            Hidden below desktop.
          </Hidden>
        </Placeholder>
      ),
    },
    {
      Code: () => (
        <Placeholder>
          The following text node is hidden above mobile:
          <Hidden above="mobile" inline={true}>
            Hidden above mobile.
          </Hidden>
        </Placeholder>
      ),
    },
    {
      label: "Hiding text nodes above tablet",
      Code: () => (
        <Placeholder>
          The following text node is hidden above tablet:
          <Hidden above="tablet" inline={true}>
            Hidden above tablet.
          </Hidden>
        </Placeholder>
      ),
    },
    {
      label: "Hiding text nodes on print",
      Code: () => (
        <Placeholder>
          The following text node is hidden on print:
          <Hidden inline={true} print={true}>
            Hidden on print.
          </Hidden>
        </Placeholder>
      ),
    },
    {
      label: "Hiding text nodes on screen readers",
      Code: () => (
        <Placeholder>
          The following text node is hidden on screen:
          <Hidden inline={true} screen={true}>
            Hidden on screen.
          </Hidden>
        </Placeholder>
      ),
    },
  ],
};
