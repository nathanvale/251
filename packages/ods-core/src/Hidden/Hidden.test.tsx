import React from "react";
import { render } from "@origin-digital/ods-testing-library";
import { Placeholder } from "../Placeholder";
import { Hidden } from "./Hidden";

test("it renders a Hidden", () => {
  const { container } = render(
    <Hidden>
      <Placeholder label="Not hidden" />
    </Hidden>
  );
  expect(container.firstChild).toMatchSnapshot();
});

test("it hides components below the md breakpoint", () => {
  const { container } = render(
    <Hidden below="md">
      <Placeholder label="Hidden below md." />
    </Hidden>
  );
  expect(container.firstChild).toMatchSnapshot();
});

test("it hides components above the sm breakpoint", () => {
  const { container } = render(
    <Hidden above="sm">
      <Placeholder label="Hidden above sm." />
    </Hidden>
  );
  expect(container.firstChild).toMatchSnapshot();
});

test("it hides text nodes below md", () => {
  const { container } = render(
    <span>
      The following text node is hidden below md:
      <Hidden below="md" inline={true}>
        Hidden below md.
      </Hidden>
    </span>
  );
  expect(container.firstChild).toMatchSnapshot();
});

test("it hides text nodes above md", () => {
  const { container } = render(
    <span>
      The following text node is hidden above md:
      <Hidden above="md" inline={true}>
        Hidden above md.
      </Hidden>
    </span>
  );
  expect(container.firstChild).toMatchSnapshot();
});
