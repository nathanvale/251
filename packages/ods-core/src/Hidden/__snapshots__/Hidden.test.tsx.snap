// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it hides components above the sm breakpoint 1`] = `
.c1 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  background-color: #e3e3e3;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-top: 16px;
  padding-bottom: 16px;
  padding-left: 24px;
  padding-right: 24px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  display: block;
}

@media screen and (min-width:576px) {
  .c0 {
    display: block;
  }
}

@media screen and (min-width:768px) {
  .c0 {
    display: none;
  }
}

@media screen and (min-width:992px) {
  .c0 {
    display: none;
  }
}

@media screen and (min-width:1200px) {
  .c0 {
    display: none;
  }
}

<div
  class="c0"
  display="block,block,none,none,none"
>
  <div
    class="c1"
    data-id="placeholder"
    display="flex"
  >
    <span
      style="font-size: 16px; line-height: 24px; font-weight: 500; color: rgb(35, 35, 35);"
    >
      Hidden above sm.
    </span>
  </div>
</div>
`;

exports[`it hides components below the md breakpoint 1`] = `
.c1 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  background-color: #e3e3e3;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-top: 16px;
  padding-bottom: 16px;
  padding-left: 24px;
  padding-right: 24px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  display: none;
}

@media screen and (min-width:576px) {
  .c0 {
    display: none;
  }
}

@media screen and (min-width:768px) {
  .c0 {
    display: block;
  }
}

@media screen and (min-width:992px) {
  .c0 {
    display: block;
  }
}

@media screen and (min-width:1200px) {
  .c0 {
    display: block;
  }
}

<div
  class="c0"
  display="none,none,block,block,block"
>
  <div
    class="c1"
    data-id="placeholder"
    display="flex"
  >
    <span
      style="font-size: 16px; line-height: 24px; font-weight: 500; color: rgb(35, 35, 35);"
    >
      Hidden below md.
    </span>
  </div>
</div>
`;

exports[`it hides text nodes above md 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  background-color: #e3e3e3;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-top: 16px;
  padding-bottom: 16px;
  padding-left: 24px;
  padding-right: 24px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  display: inline;
}

@media screen and (min-width:576px) {
  .c1 {
    display: inline;
  }
}

@media screen and (min-width:768px) {
  .c1 {
    display: inline;
  }
}

@media screen and (min-width:992px) {
  .c1 {
    display: none;
  }
}

@media screen and (min-width:1200px) {
  .c1 {
    display: none;
  }
}

<div
  class="c0"
  data-id="placeholder"
  display="flex"
>
  <span
    style="font-size: 16px; line-height: 24px; font-weight: 500; color: rgb(35, 35, 35);"
  >
    The following text node is hidden above md:
    <span
      class="-span c1"
      display="inline,inline,inline,none,none"
    >
      Hidden above md.
    </span>
  </span>
</div>
`;

exports[`it hides text nodes below md 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  background-color: #e3e3e3;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-top: 16px;
  padding-bottom: 16px;
  padding-left: 24px;
  padding-right: 24px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  display: none;
}

@media screen and (min-width:576px) {
  .c1 {
    display: none;
  }
}

@media screen and (min-width:768px) {
  .c1 {
    display: inline;
  }
}

@media screen and (min-width:992px) {
  .c1 {
    display: inline;
  }
}

@media screen and (min-width:1200px) {
  .c1 {
    display: inline;
  }
}

<div
  class="c0"
  data-id="placeholder"
  display="flex"
>
  <span
    style="font-size: 16px; line-height: 24px; font-weight: 500; color: rgb(35, 35, 35);"
  >
    The following text node is hidden below md:
    <span
      class="-span c1"
      display="none,none,inline,inline,inline"
    >
      Hidden below md.
    </span>
  </span>
</div>
`;

exports[`it renders a Hidden 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  display: block;
}

.c1 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  background-color: #e3e3e3;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-top: 16px;
  padding-bottom: 16px;
  padding-left: 24px;
  padding-right: 24px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

@media screen and (min-width:576px) {
  .c0 {
    display: block;
  }
}

@media screen and (min-width:768px) {
  .c0 {
    display: block;
  }
}

@media screen and (min-width:992px) {
  .c0 {
    display: block;
  }
}

@media screen and (min-width:1200px) {
  .c0 {
    display: block;
  }
}

<div
  class="c0"
  display="block,block,block,block,block"
>
  <div
    class="c1"
    data-id="placeholder"
    display="flex"
  >
    <span
      style="font-size: 16px; line-height: 24px; font-weight: 500; color: rgb(35, 35, 35);"
    >
      Not hidden
    </span>
  </div>
</div>
`;
